portal-impl/src/com/liferay/portal/db/partition/DBPartitionUtil.java

```diff
@@ -272,6 +275,14 @@ public static DataSource wrapDataSource(DataSource dataSource)
 
 		try (Connection connection = dataSource.getConnection()) {
 			_defaultSchemaName = connection.getCatalog();
+
```

Comment:
@LuisOrtiz89 @achaparro @marianoalvarosaiz this is too hacky. It basically creates a "duplicated" way to populate PortalInstancePool while the original way still exists which will eventually overwrite the map with "same" values.

This should be sorted out as an unified startup sequence, either wait for the PortalInstancePool population or do a catch up when the companies are eventually registered into PortalInstancePool. Duplicating the execution paths here is a very bad design, and very likely to hurt us back in the future with some tricky bugs.